### PR TITLE
Refactor locators.json handling to use common loader

### PR DESCRIPTION
**Pull request by avaissi**
- _get_locators_json_info() and _remove_locator_from_json_internal() to use shared loader for LocatorsDatabase

### CODE REVIEW

File: robocorp-code/vscode-client/src/views.ts [hunk 1]
```diff
@@ -229,13 +229,7 @@ export class LocatorsTreeDataProvider implements vscode.TreeDataProvider<Locator
...
         if (!actionResult['success']) {
             this.lastRobotEntry = undefined;
-            return [{
-                name: actionResult.message,
-                type: "error",
-                line: 0,
-                column: 0,
-                filePath: robotEntry.robot.filePath,
-            }];
+            return [];
```
Comment (fabioz) [inline robocorp-code/vscode-client/src/views.ts:232]: Why this change? 
-- i.e.: if there's some error loading the info, I think it's reasonable to show it to users...

File: robocorp-code/src/robocorp_code/robocorp_language_server.py [hunk 2]
```diff
@@ -990,55 +1015,41 @@ def _get_locators_json_info(
...
         locators_json_info: List[LocatorEntryInfoDict] = []
         locator: Locator
-        if locators_json.exists():
+        db, locators_json = None, None
+        try:
+            action_result: ActionResultDictLocatorsJson = self._load_locators_db(path)
+            if action_result["success"]:
+                db, locators_json = action_result["result"]
+            else:
+                raise Exception(action_result["message"])
```
Comment (fabioz) [inline robocorp-code/src/robocorp_code/robocorp_language_server.py:1026]: This seems a bit odd for me... 

i.e.: Since `ActionResultDictLocatorsJsonInfo` has a value indicating the success, I think you should just return one with a `success: False` (otherwise it could just return a `List[LocatorEntryInfoDict]` and throw exceptions).

File: robocorp-code/src/robocorp_code/robocorp_language_server.py [hunk 3]
```diff
@@ -990,55 +1015,41 @@ def _get_locators_json_info(
...
+                    {
+                        "name": name,
+                        "line": line,
+                        "column": col,
+                        "type": as_dict["type"],
+                        "filePath": str(locators_json),
+                    }
+                )
+        except Exception as e:
+            return {"success": False, "message": str(e), "result": None}
```
Comment (fabioz) [inline robocorp-code/src/robocorp_code/robocorp_language_server.py:1045]: Exceptions should be logged (i.e.: `log.exception("Error message")`).

**Review by fabioz — CHANGES_REQUESTED** [review]

File: robocorp-code/vscode-client/src/views.ts (hunk 1 above)
Comment (avaissi) [reply robocorp-code/vscode-client/src/views.ts:232]: This populates the Locators view with unnecessary tree items, even in "normal" case of not having a locators.json for the robot, or an empty locators.json. Might be better to show errors as notifications maybe?

Comment (fabioz) [reply robocorp-code/vscode-client/src/views.ts:232]: I'd say that if we want to hide and count that as "normal", the language server should return `success:true` with an empty list (so that errors are still shown) and still show errors here.

Comment (avaissi) [reply robocorp-code/vscode-client/src/views.ts:232]: That sounds fair enough, I'll modify those cases so that the language server to respond with success and an empty list

Comment (avaissi) [reply robocorp-code/vscode-client/src/views.ts]: Handling now the empty or missing locators.json in language server and showing any errors to user as before

File: robocorp-code/src/robocorp_code/robocorp_language_server.py (hunk 2 above)
Comment (avaissi) [reply robocorp-code/src/robocorp_code/robocorp_language_server.py]: Modified the if-else to return rather than raise Exception

File: robocorp-code/src/robocorp_code/robocorp_language_server.py (hunk 3 above)
Comment (avaissi) [reply robocorp-code/src/robocorp_code/robocorp_language_server.py:1038]: Added log.exception()

File: robocorp-code/bin/create_env/conda.yaml [hunk 4]
```diff
@@ -1,10 +1,9 @@
 channels:
-  - defaults
   - conda-forge
 dependencies:
-  - python=3.8
+  - python=3.8.10
```
Comment (avaissi) [inline robocorp-code/bin/create_env/conda.yaml:4]: Pinned the python version to avoid faulty version greeping

File: robocorp-code/src/robocorp_code/robocorp_language_server.py [hunk 5]
```diff
@@ -1048,28 +1052,18 @@ def _remove_locator_from_json_internal(
...
-                            try:
-                                error = error[0] % error[1]
-                            except:
-                                error = str(error)
-                        else:
-                            error = str(error)
-
-                    return {"success": False, "message": error, "result": None}
+        db, locators_json = None, None
+        action_result: ActionResultDictLocatorsJson = self._load_locators_db(path)
```
Comment (fabioz) [inline robocorp-code/src/robocorp_code/robocorp_language_server.py:1056]: I think that this line should be inside the `try..except` below as it can raise an exception when loading the db.

File: robocorp-code/src/robocorp_code/robocorp_language_server.py [hunk 6]
```diff
@@ -1048,28 +1052,18 @@ def _remove_locator_from_json_internal(
...
+        try:
+            if action_result["success"]:
+                db, locators_json = action_result["result"]
+            else:
+                raise Exception(action_result["message"])
```
Comment (fabioz) [inline robocorp-code/src/robocorp_code/robocorp_language_server.py:1061]: Instead of raising the exception it could be just:

return {"success": False, "message": action_result["message"], "result": None}

File: robocorp-code/vscode-client/src/views.ts [hunk 7]
```diff
@@ -1,12 +1,27 @@
-import { TREE_VIEW_ROBOCORP_CLOUD_TREE, TREE_VIEW_ROBOCORP_LOCATORS_TREE, TREE_VIEW_ROBOCORP_ROBOTS_TREE, TREE_VIEW_ROBOCORP_ROBOT_CONTENT_TREE } from './robocorpViews';
+import {
```
Comment (fabioz) [inline robocorp-code/vscode-client/src/views.ts:1]: Note: we should probably settle into some formatter for the future so that imports aren't all reformatted all the time -- no need to change anything right now, just a reminder to myself while notifying you too ;)

File: robocorp-code/vscode-client/src/locators.ts [hunk 8]
```diff
@@ -8,25 +13,41 @@ import { OUTPUT_CHANNEL } from './channel';
...
     let locatorSelected: LocatorEntry | undefined = locator || getSelectedLocator();
     if (locatorSelected) {
-        vscode.env.clipboard.writeText(locatorSelected.name);
+        env.clipboard.writeText(locatorSelected.name);
     }
 }
 
 export async function removeLocator(locator?: LocatorEntry) {
+    // Confirmation dialog button texts
+    const DELETE = 'Delete';
```
Comment (fabioz) [inline robocorp-code/vscode-client/src/locators.ts:22]: Instead of `getSelectedLocator()`, shouldn't it use the one received as a parameter (if available)?

File: robocorp-code/vscode-client/src/locators.ts [hunk 9]
```diff
@@ -8,25 +13,41 @@ import { OUTPUT_CHANNEL } from './channel';
 export async function copySelectedToClipboard(locator?: LocatorEntry) {
     let locatorSelected: LocatorEntry | undefined = locator || getSelectedLocator();
     if (locatorSelected) {
-        vscode.env.clipboard.writeText(locatorSelected.name);
+        env.clipboard.writeText(locatorSelected.name);
```
Comment (fabioz) [inline robocorp-code/vscode-client/src/locators.ts:16]: Instead of `getSelectedLocator()`, shouldn't it use the one received as a parameter (if available)?

**Review by fabioz — CHANGES_REQUESTED** [review]

File: robocorp-code/vscode-client/src/views.ts (hunk 7 above)
Comment (avaissi) [reply robocorp-code/vscode-client/src/views.ts:1]: Right, that makes sense. Maybe we should config a prettier/linter configs for python and typescript/javascript as well, if not already there.. I've tried to remember to format python with black at least :)

File: robocorp-code/vscode-client/src/locators.ts (hunk 8 above)
Comment (avaissi) [reply robocorp-code/vscode-client/src/locators.ts:22]: let locatorSelected: LocatorEntry | undefined = locator || getSelectedLocator();
The || notation should take care of prioritizing the parameter value.. could be ?? as well to strictly compare it with undefined (optional value)

File: robocorp-code/vscode-client/src/locators.ts (hunk 9 above)
Comment (avaissi) [reply robocorp-code/vscode-client/src/locators.ts:16]: This might be just a bit poor choice of variable name for the `locatorSelected` . The previous comment applies here as well

Comment (fabioz) [reply robocorp-code/vscode-client/src/locators.ts:16]: You're right, sorry, my bad upon reviewing.

File: robocorp-code/vscode-client/src/locators.ts (hunk 8 above)
Comment (fabioz) [reply robocorp-code/vscode-client/src/locators.ts:22]: You're right, sorry, my bad upon reviewing.

File: robocorp-code/src/robocorp_code/robocorp_language_server.py (hunk 6 above)
Comment (avaissi) [reply robocorp-code/src/robocorp_code/robocorp_language_server.py:1061]: Damn, missed this with the `raise Exception` changes to returns.. fixing

File: robocorp-code/src/robocorp_code/robocorp_language_server.py (hunk 5 above)
Comment (avaissi) [reply robocorp-code/src/robocorp_code/robocorp_language_server.py]: added inside try..except
